pkg/engine/renderer/enhancer.go

```diff
@@ -75,12 +73,12 @@ func (k *KustomizeEnhancer) Apply(templates map[string]string, metadata Metadata
 
 	yamlBytes, err := yaml.Marshal(kustomization)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error marshalling kustomize yaml")
+		return nil, fmt.Errorf("error marshalling kustomize yaml: %w", err)
```

Comment:
Do we have an agreement on when we expose the underlying error with `%w` and when not? 

To quote Go 1.13 [errors blog post](https://blog.golang.org/go1.13-errors):
```
In other words, wrapping an error makes that error part of your API. If you don't want to 
commit to supporting that error as part of your API in the future, you shouldn't wrap the error.

It’s important to remember that whether you wrap or not, the error text will be the same. A 
person trying to understand the error will have the same information either way; the choice to 
wrap is about whether to give programs additional information so they can make more informed 
decisions, or to withhold that information to preserve an abstraction layer.
```

I believe, in this particular case, it doesn't make a difference since our `yaml.Marshal` method already wraps underlying errors. Personally, I'm leaning towards:
- Using `%v` by default to preserve abstractions boundaries
- Using `%w` intentionally when exposing the underlying error type is necessary

Thoughts?

Reply:
It seems we're already having this discussion in [this PR](https://github.com/kudobuilder/kudo/pull/1104#discussion_r350606507). I x-posted my comment there.

Reply:
great thoughts.   The wrapping is because we wrapped it before... I'm not aware if this was incidental or intentional.   It is worth pausing..    I definitely like your personally leanings and will take them on as well.